pkg/apis/core/validation/validation.go

```diff
@@ -1154,6 +1154,71 @@ func validateProjectionSources(projection *core.ProjectedVolumeSource, projectio
 				allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 			}
 		}
+		if projPath := srcPath.Child("clusterTrustBundlePEM"); source.ClusterTrustBundle != nil {
+			numSources++
+
+			usingName := source.ClusterTrustBundle.Name != nil
+			usingSignerName := source.ClusterTrustBundle.SignerName != nil
+
+			switch {
+			case usingName && usingSignerName:
+				allErrs = append(allErrs, field.Invalid(projPath, source.ClusterTrustBundle, "only one of name and signerName may be used"))
+			case usingName:
+				if *source.ClusterTrustBundle.Name == "" {
+					allErrs = append(allErrs, field.Required(projPath.Child("name"), "must be a valid object name"))
+				}
+
+				name := *source.ClusterTrustBundle.Name
+				if splitPoint := strings.LastIndex(name, ":"); splitPoint != -1 {
+					// This looks like it refers to a signerName trustbundle.
+					signerName := strings.ReplaceAll(name[:splitPoint], ":", "/")
+					if signerNameErrs := ValidateSignerName(projPath.Child("name"), signerName); len(signerNameErrs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+
+					additionalName := name[splitPoint+1:]
+					if errMsgs := apimachineryvalidation.NameIsDNSSubdomain(additionalName, false); len(errMsgs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+				} else {
+					// This refers to a non-signerName trustbundle
+					if errMsgs := apimachineryvalidation.NameIsDNSSubdomain(name, false); len(errMsgs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+				}
+
+				if source.ClusterTrustBundle.LabelSelector != nil {
+					allErrs = append(allErrs, field.Invalid(projPath.Child("labelSelector"), source.ClusterTrustBundle.LabelSelector, "labelSelector must be nil if name is specified"))
+				}
+			case usingSignerName:
+				if *source.ClusterTrustBundle.SignerName == "" {
+					allErrs = append(allErrs, field.Required(projPath.Child("signerName"), "must be a valid signer name"))
+				}
+
+				allErrs = append(allErrs, ValidateSignerName(projPath.Child("signerName"), *source.ClusterTrustBundle.SignerName)...)
+
+				labelSelectorErrs := unversionedvalidation.ValidateLabelSelector(
+					source.ClusterTrustBundle.LabelSelector,
+					unversionedvalidation.LabelSelectorValidationOptions{AllowInvalidLabelValueInSelector: false},
+					projPath.Child("labelSelector"),
+				)
+				allErrs = append(allErrs, labelSelectorErrs...)
+
+			default:
+				allErrs = append(allErrs, field.Required(projPath, "either name or signerName must be specified"))
+			}
+
+			if source.ClusterTrustBundle.Path == "" {
+				allErrs = append(allErrs, field.Required(projPath.Child("path"), ""))
+			}
+
```

Comment:
also call `validateLocalNonReservedPath` on `source.ClusterTrustBundle.Path` to catch paths the kubelet will reject (like absolute paths, paths containing backsteps, etc): https://github.com/kubernetes/kubernetes/blob/f7cb8a5e8a860df643e143cde54d34080372b771/pkg/volume/util/atomic_writer.go#L252-L264


configmap, secret, and downwardAPI already call this. projected token does not (@enj, can you open an issue to ratchet validate that as well)